step-framework-parent/step-framework-server/src/main/java/step/framework/server/tables/service/TableService.java

```diff
@@ -131,8 +132,8 @@ public <T extends AbstractIdentifiableObject> TableBulkOperationReport performBu
             if (targetType == TableBulkOperationTargetType.FILTER || targetType == TableBulkOperationTargetType.ALL) {
                 Filter filter = createFilter(parameters, session, table);
                 Collection<T> collection = table.getCollection();
-                collection.find(filter, null, null, null, 0).map(e -> e.getId().toString())
-                        .forEach(id -> countingOperationById.accept(id, parameters.isPreview()));
+                List<String> ids = collection.find(filter, null, null, null, 0).map(e -> e.getId().toString()).collect(Collectors.toList());
```

Comment:
Would add a comment to explain why the list of ids cannot be streamed in order to avoid refactorings in the future

Reply:
comment added